application/controllers/admin/Database.php

```diff
@@ -657,6 +657,20 @@ private function actionUpdateSurveyLocaleSettings($iSurveyID)
             // Save plugin settings : actually leave it before saving core : we are sure core settings is saved in LS way.
             $pluginSettings = App()->request->getPost('plugin', array());
             foreach ($pluginSettings as $plugin => $settings) {
+                // Call plugin to check and fix, do not force reload
+                $oPlugin = App()->getPluginManager()->loadPlugin($plugin, null, false);
+                if ($oPlugin) {
+                    /* Fix settings for datetime **/
+                    foreach ($settings as $setting => $value) {
+                        if (is_array($value) && !empty($value['datetime']) && !empty($value['datetimesaveformat'])) {
```

Comment:
Shouldn't we move this date processing logic to a method/function?
If not, this loop will easily get hard to read.
Also, easier to test with test script

Reply:
Don't recall that much now, but, on the if statement, shouldn't we also check the setting type equals date?

Reply:
> Shouldn't we move this date processing logic to a method/function? If not, this loop will easily get hard to read. Also, easier to test with test script

Yes, i know, but password have same issue … https://github.com/LimeSurvey/LimeSurvey/pull/2753#issuecomment-1333844087

Since the 2 files pull request update same file i want to minimize the difference.

When both are merged : maybe a new function in PluginBase ?

The big issue : all settings are managed only via frontend, nothing is done currently on backend. Then : we can not really fix via core PHP …

Reply:
I agree with Gabriel. Should be moved out to separate function to avoid duplication and too long functions.